.github/workflows/lint.yaml

```diff
@@ -72,11 +72,24 @@ jobs:
         # https://taplo.tamasfe.dev/cli/installation/binary.html
         run: curl -fsSL https://github.com/tamasfe/taplo/releases/download/0.10.0/taplo-linux-x86_64.gz | gzip -d - | install -m 755 /dev/stdin /usr/local/bin/taplo
 
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install markdown-toc
+        run: npm install -g markdown-toc
+
       - name: Verify files format using markdownlint-cli2
         uses: DavidAnson/markdownlint-cli2-action@v20
         with:
           config: .markdownlint-cli2.yaml
 
+      - name: Verify TOC in markdown files
+        run: |
+          markdown-toc --maxdepth 4 --no-first1 --bullets "-" -i README.md && git diff --exit-code README.md
+          markdown-toc --maxdepth 2 --no-first1 --bullets "-" -i RULES_DESCRIPTIONS.md && git diff --exit-code RULES_DESCRIPTIONS.md
+
```

Comment:
We should consider splitting this huge jobs into smaller one maybe.

Because here, it's not clear why node.js is installed.

But this can be done in a separate PR